### ISSUE TITLE
Make it look nice

### ISSUE THREAD

**Issue by tombreit** [tombreit/mkdocs-pagetree-plugin#3]
... and fix `<details>` when Material for Mkdocs is used.

### MkDocs builtin default theme

![grafik](https://github.com/user-attachments/assets/f96dfd19-3fc2-4dd4-8c05-6af83d3675aa)

### Material for MkDocs theme

![grafik](https://github.com/user-attachments/assets/c52cbe20-1f6c-4197-b897-2090cf151822)

**Comment by SuibianP** [comment]
By default `<details>` is rendered on mkdocs-material as a collapsed admonition box.

**Comment by Tanikai** [comment]
I am currently working on improving the style. I have:

- Removed the bullet points for expandable nodes, and
- Aligned the symbols of leaf and expandable nodes that are on the same hierarchy level.

This design was inspired by the page tree macro from Confluence ([source](https://confluence.atlassian.com/doc/files/163414255/1005333658/1/1589961908269/page-tree-params.png)):
![page-tree-params](https://github.com/user-attachments/assets/0bece73d-d328-432e-8e4a-e5d7afba1886)


Here are some screenshots:

## MkDocs default

<img width="513" alt="Screenshot 2024-12-23 at 23 51 39" src="https://github.com/user-attachments/assets/4fa802f0-4632-4c21-8bb1-f0831298898f" />

In Confluence, every hierachy level uses the same "disc" symbol. I think that this might be suitable for here as well.

## Material for MkDocs

<img width="536" alt="Screenshot 2024-12-23 at 23 51 25" src="https://github.com/user-attachments/assets/dd0b14d6-7308-4cf1-a95c-200329264a28" />

The alignment is a bit off starting at the second level, I'm guessing this has something to do with the font size changing. I'll look a bit into this in the following days.


What do you think about this styling approach? Do you have any suggestions / ideas regarding the styling?

**Comment by tombreit** [comment]
Looks good 👍

**Comment by tombreit** [comment]
Hi @Tanikai ,

I've "unified" the styling across different themes (builtin MkDocs and ReadTheDocs themes, Material for MkDocs theme), using "caret-style" toggle icons:

### Theme: MkDocs

<img width=300 src="https://github.com/user-attachments/assets/1ba32d01-057b-4236-a87e-e2120cb15681" alt="">

### Theme: ReadTheDocs

<img width=300 src="https://github.com/user-attachments/assets/acd10793-0435-44bc-b636-967fb2ac3446" alt="">

### Theme: Material for MkDocs

<img width=300 src="https://github.com/user-attachments/assets/2aafd16c-a1ac-4d5b-abdc-e2439fc00f75" alt="">

<hr>

You can review/test this in the following branch:  
https://github.com/tombreit/mkdocs-pagetree-plugin/compare/caret-style

**Comment by Tanikai** [comment]
Looks good! There is a small bug in Safari where the default triangles are shown:

<img width="432" alt="Screenshot 2024-12-28 at 12 18 04" src="https://github.com/user-attachments/assets/531df25a-a127-4776-9971-9bcfe5ff54fb" />

You can hide them with the following CSS code:

```css
.pagetree summary::-webkit-details-marker {
  display: none;
}
```

Here are screenshots for the Default, ReadTheDocs and Material theme in Safari after the fix:

<img width="447" alt="image" src="https://github.com/user-attachments/assets/fafe0b2f-334d-4a91-a02e-7dc5799a3926" />

<img width="489" alt="image" src="https://github.com/user-attachments/assets/b9c99234-1973-45ba-a857-f2361fce543f" />

<img width="428" alt="image" src="https://github.com/user-attachments/assets/851a8b4d-7854-4b9c-8dd8-80278b2eabf8" />

**Comment by tombreit** [comment]
Thanks for the Safari fix, it is now included. Could also test it locally with a WebKit-based browser on Linux. “My” WebKit-based browser unfortunately did not get the alignment/positioning of the bullet points correct.

I have also changed the CSS again, as I also want to support the *ReadTheDocs* theme. Overall, however, I'm not happy with the current CSS, especially with my overuse of `!important`. 
But since the three themes I would like to support (at least) (`mkdocs`, `readthedocs`, `material`) don't make it easy either, it's at least a start.

**Comment by tombreit** [comment]
Hi @Tanikai , I would like to close this issue, it was/is too broad anyway and I think we fixed the issues you mentioned (remove bullet points if a details toggle exists, fixed indentation).

**Comment by Tanikai** [comment]
Yes, I agree. I think we did a good job of making it look better than before.
